common/schema/validate.go

```diff
@@ -57,10 +57,26 @@ func (s *Schema) Validate() (*Schema, error) {
 
 // ValidateModuleInSchema clones and normalises a schema and semantically validates a single module within it.
 // If no module is provided, all modules in the schema are validated.
+// m can be a new or updated module that will be added to the schema before validation.
 //
 //nolint:maintidx
 func ValidateModuleInSchema(schema *Schema, m optional.Option[*Module]) (*Schema, error) {
```

Comment:
Previously this was not actually validating `m` but rather the matching module in `schema` for its name.
If we wanted that we should probably just pass an optional name into this func.

But also to achieve what we want in this PR it is very convenient to just pass in the module we want to validate with the schema and have it all work.